lib/instrumentation/modules/http2.js

```diff
@@ -0,0 +1,86 @@
+'use strict'
+
+var debug = require('debug')('elastic-apm')
+var eos = require('end-of-stream')
+var shimmer = require('../shimmer')
+
+module.exports = function (http2, agent) {
+  debug('shimming http2.createServer function')
+  shimmer.wrap(http2, 'createServer', wrapCreateServer)
+  shimmer.wrap(http2, 'createSecureServer', wrapCreateServer)
+
+  return http2
+
+  // The `createServer` function will unpatch itself after patching
+  // the first server prototype it patches.
+  function wrapCreateServer (original) {
+    return function wrappedCreateServer (options, handler) {
+      var server = original.apply(this, arguments)
+      shimmer.wrap(server.constructor.prototype, 'emit', wrapEmit)
+      wrappedCreateServer.__obUnwrap()
+      return server
+    }
+  }
+
+  function wrapEmit (original) {
+    var patched = false
+    return function wrappedEmit (event, stream, headers) {
+      if (event === 'stream') {
+        if (!patched) {
+          patched = true
+          shimmer.wrap(stream.constructor.prototype, 'respond', wrapRespond)
+          shimmer.wrap(stream.constructor.prototype, 'end', wrapEnd)
+        }
+
+        // TODO: Figure out if we need to build a response object too
```

Comment:
@Qard Is this TODO still relevant? Seems like you build a response object below

Reply:
fixed